src/components/CompareResults/ResultsTable.tsx

```diff
@@ -67,10 +68,13 @@ const customStyles = {
   background: 'none',
 };
 
-function ResultsTable() {
+function ResultsTable(props: { isOverTime: boolean }) {
   const themeMode = useAppSelector((state) => state.theme.mode);
+  const { isOverTime } = props;
+  const { results: withBaseResults } = useLoaderData() as LoaderReturnValue;
+  const { results: overTimeResults } = useLoaderData() as OverTimeLoadResults;
```

Comment:
My understanding is that the only reason you need the `isOvertime` prop is because of the different types. But the `results` property should be the same type, right? So maybe we can do something like:
```
  const { results } = useLoaderData() as { results: LoaderReturnValue['results'] };
```
Typescript doesn't care because for Typescript `useLoaderData()` is not really typed properly, so we can do whatever we want, as long as we do not lie.

I believe that with this change you can remove this new prop and simplify a bunch of code.